tests/integration/submissionworker/test_submission_worker.py

```diff
@@ -0,0 +1,512 @@
+import mock
+import os
+import shutil
+import tempfile
+import time
+import json
+import zipfile
+
+from allauth.account.models import EmailAddress
+
+from datetime import timedelta
+
+from os.path import join
+
+from django.core.files.base import ContentFile
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.core.urlresolvers import reverse_lazy
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+from rest_framework.test import APITestCase, APIClient
+
+from challenges.models import (
+    Challenge,
+    ChallengePhase,
+    ChallengePhaseSplit,
+    DatasetSplit,
+    Leaderboard,
+    LeaderboardData,
+)
+from participants.models import Participant, ParticipantTeam
+from hosts.models import ChallengeHostTeam
+from jobs.models import Submission
+
+import scripts.workers.submission_worker as submission_worker
+
+
+class BaseAPITestClass(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient(enforce_csrf_checks=True)
+
+        self.user = User.objects.create(
+            username='someuser',
+            email="user@test.com",
+            password='secret_password')
+
+        EmailAddress.objects.create(
+            user=self.user,
+            email='user@test.com',
+            primary=True,
+            verified=True)
+
+        self.user1 = User.objects.create(
+            username='someuser1',
+            email="user1@test.com",
+            password='secret_password1')
+
+        EmailAddress.objects.create(
+            user=self.user1,
+            email='user1@test.com',
+            primary=True,
+            verified=True)
+
+        self.challenge_host_team = ChallengeHostTeam.objects.create(
+            team_name='Test Challenge Host Team',
+            created_by=self.user)
+
+        self.participant_team = ParticipantTeam.objects.create(
+            team_name='Participant Team for Challenge',
+            created_by=self.user1)
+
+        self.participant = Participant.objects.create(
+            user=self.user1,
+            status=Participant.SELF,
+            team=self.participant_team)
+
+        self.BASE_TEMP_DIR = tempfile.mkdtemp()
+
+        f = open(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), "x")
+        f = open(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), "w")
```

Comment:
Wouldn't this be overridden? why do we need to open the file in `execute` mode then override it again?